test/unit/utils.html

```diff
@@ -91,12 +90,76 @@
 
       setTimeout(function() {
         assert.equal(called, 1, 'debounce should be called exactly once');
-        assert(Date.now() - now > 100, 'debounce should be called after at least 100ms');
```

Comment:
In my opinion testing of ``setTimeout`` should be done by the browser vendors and in case of Microtasks in [test/unit/async.html](https://github.com/Polymer/polymer/blob/master/test/unit/async.html), so that we can rely on them here.

@kevinpschaaf thanks for pointing out that ```wait``` and ```no-wait``` debounce behavior should be tested for different issues. Added more explicit tests for each of them and as replacement of the ``Date.now`` assertion I removed there now is a second timeout after ``50ms`` that asserts the callback not to be fired yet.